deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -265,11 +265,13 @@ class ChatViewController: UITableViewController {
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         AppStateRestorer.shared.storeLastActiveChat(chatId: chatId)
+
         // things that do not affect the chatview
         // and are delayed after the view is displayed
-        dcContext.marknoticedChat(chatId: chatId)
-        let array = dcContext.getFreshMessages()
-        UIApplication.shared.applicationIconBadgeNumber = array.count
+        DispatchQueue.global(qos: .background).async { [weak self] in
+            guard let self = self else { return }
+            self.dcContext.marknoticedChat(chatId: self.chatId)
```

Comment:
you can use self? here, too

Reply:
that would result in `self?.dcContext.marknoticedChat(chatId: self?.chatId ?? 0)` - i do not think, this is a big win in readablility.

Reply:
You're right I oversaw the second self there